Author: Bernhard M. Wiedemann <bwiedemann suse de>
Date: 2021-04-01

Do not build code using instructions because they are available on a random build worker.

Instead we pin the CPU type to a reasonable compromise
to allow people to run it on >50% of systems,
but also to use CPU instructions available since 10+ years
to speed up processing.

diff --git a/setup.py b/setup.py
index c00d331..35542b6 100644
--- a/setup.py
+++ b/setup.py
@@ -168,9 +168,9 @@ class CythonBuildExt(build_ext):
         # package is being built locally, this is desired
         if not ("AZURE_BUILD" in os.environ or "CONDA_BUILD" in os.environ):
             if platform.machine() == "ppc64le":
-                extra_compile_args += ["-mcpu=native"]
+                extra_compile_args += ["-mcpu=power8"]
             if platform.machine() == "x86_64":
-                extra_compile_args += ["-march=native"]
+                extra_compile_args += ["-march=corei7"]
 
         # We will disable openmp flags if the compiler doesn"t support it. This
         # is only really an issue with OSX clang
